packages/dx-react-grid-demos/src/demo-sources/grid-featured-data-editing/bootstrap3/demo.jsx

```diff
@@ -137,189 +137,158 @@ const EditCell = (props) => {
+    { columnName: 'customer', width: 200 },
+  ]);
+  const [sorting, getSorting] = useState([]);
+  const [editingRowIds, getEditingRowIds] = useState([]);
+  const [addedRows, setAddedRows] = useState([]);
+  const [rowChanges, setRowChanges] = useState({});
+  const [currentPage, setCurrentPage] = useState(0);
+  const [pageSize, setPageSize] = useState(0);
+  const [pageSizes] = useState([5, 10, 0]);
+  const [columnOrder, setColumnOrder] = useState(['product', 'region', 'amount', 'discount', 'saleDate', 'customer']);
+  const [currencyColumns] = useState(['amount']);
+  const [percentColumns] = useState(['discount']);
+  const [leftFixedColumns] = useState([TableEditColumn.COLUMN_TYPE]);
+  const [totalSummaryItems] = useState([
+    { columnName: 'discount', type: 'avg' },
+    { columnName: 'amount', type: 'sum' },
+  ]);
 
-    this.state = {
-      columns: [
-        { name: 'product', title: 'Product' },
-        { name: 'region', title: 'Region' },
-        { name: 'amount', title: 'Sale Amount' },
-        { name: 'discount', title: 'Discount' },
-        { name: 'saleDate', title: 'Sale Date' },
-        { name: 'customer', title: 'Customer' },
-      ],
-      tableColumnExtensions: [
-        { columnName: 'product', width: 200 },
-        { columnName: 'region', width: 180 },
-        { columnName: 'amount', width: 180, align: 'right' },
-        { columnName: 'discount', width: 180 },
-        { columnName: 'saleDate', width: 180 },
-        { columnName: 'customer', width: 200 },
-      ],
-      rows: generateRows({
-        columnValues: { id: ({ index }) => index, ...globalSalesValues },
-        length: 12,
-      }),
-      sorting: [],
-      editingRowIds: [],
-      addedRows: [],
-      rowChanges: {},
-      currentPage: 0,
-      pageSize: 0,
-      pageSizes: [5, 10, 0],
-      columnOrder: ['product', 'region', 'amount', 'discount', 'saleDate', 'customer'],
-      currencyColumns: ['amount'],
-      percentColumns: ['discount'],
-      leftFixedColumns: [TableEditColumn.COLUMN_TYPE],
-      totalSummaryItems: [
-        { columnName: 'discount', type: 'avg' },
-        { columnName: 'amount', type: 'sum' },
-      ],
-    };
-
-    const getStateRows = () => {
-      const { rows } = this.state;
-      return rows;
-    };
+  const changeSorting = value => getSorting(value);
+  const changeEditingRowIds = value => getEditingRowIds(value);
+  const changeAddedRows = value => setAddedRows(
+    value.map(row => (Object.keys(row).length ? row : {
+      amount: 0,
+      discount: 0,
+      saleDate: new Date().toISOString().split('T')[0],
+      product: availableValues.product[0],
+      region: availableValues.region[0],
+      customer: availableValues.customer[0],
+    })),
+  );
+  const changeRowChanges = value => setRowChanges(value);
+  const changeCurrentPage = value => setCurrentPage(value);
+  const changePageSize = value => setPageSize(value);
+  const changeColumnOrder = value => setColumnOrder(value);
```

Comment:
Why we create new functions instead of using `setXXX` functions from `useState`?

Reply:
In controlled mode we must able to do few actions on change state. If we will use `setXXX`, we can't do it.

Reply:
Why can't we? `changePageSize` does exactly the same as `setRowChanges`

Reply:
In this case yes. But it's wont working if we will want to do few actions, like in `grid-searching remote-mode` demo
```
const changeSearchValue = (value) => {
    setLoading(true);
    setSearchValue(value);
  };
```